ui/src/utils/ui.ts

```diff
@@ -390,3 +390,12 @@ export function getElementCoordinates(element: HTMLElement) {
     clientHeight: element.clientHeight,
   };
 }
+
+/**
+ * TODO(andrewhead): Handle the case of arXiv publications that have multiple versions. How do we
+ * make sure we're querying for the same version of paper data as the paper that was opened?
+ */
+ export function extractArxivId(url: string): string | undefined {
+  const matches = url.match(/arxiv\.org\/pdf\/(.*)(?:\.pdf).*$/) || [];
```

Comment:
If you change this to the `.exec()` API, then it won't fail if the string is `undefined`/`null`.
```
const matches = /arxiv\.org\/pdf\/(.*)(?:\.pdf).*$/.exec(url) || [];
```

Also, don't need the `.*$` part in the regex, because `.*` means any chars and `$` means end of the string.

If you have access to the `url` library, you could do this too:
```
import URL from 'url';
...
const urlObj = new URL(urlStr);
const matches = /^\/pdf\/(.*)(?:\.pdf$)/.exec(urlObj.pathname) || [];
```

That would make sure it's a valid URL, and that the pathname ends in `.pdf` (and not `.pdf.zip` or something). It'll strip off the query string for you.

Reply:
Cool, yeah I've confirmed that the final `.*$` is no longer needed (I think the real bug in the regex must've been making the .pdf extension optional).